client/src/components/OptionsDrawer/OptionsDrawer.tsx

```diff
@@ -1,23 +1,39 @@
-import Typography from '@material-ui/core/Typography';
-import ListItem from '@material-ui/core/ListItem';
-import Divider from '@material-ui/core/Divider';
-import List from '@material-ui/core/List';
-import Drawer from '@material-ui/core/Drawer';
-import React from 'react';
-import ListItemText from '@material-ui/core/ListItemText';
-import ListItemIcon from '@material-ui/core/ListItemIcon';
+import { useState, useEffect, Dispatch, SetStateAction } from 'react';
+import { Typography, ListItem, ListItemText, ListItemIcon, List, Divider, Drawer } from '@material-ui/core';
 import SwapHorizontalCircleOutlinedIcon from '@material-ui/icons/SwapHorizontalCircleOutlined';
 import useStyles from './optionsDrawerStyles';
+import { getUserBoards, getBoard } from '../../helpers/';
+import { useKanban } from '../../context/useKanbanContext';
 import { IBoard } from '../../interface/Board';
 
 interface Props {
   open: boolean;
-  boards: Array<IBoard>;
-  setOpen: React.Dispatch<React.SetStateAction<boolean>>;
+  setOpen: Dispatch<SetStateAction<boolean>>;
 }
 
-const OptionsDrawer = ({ open, boards, setOpen }: Props): JSX.Element => {
+const OptionsDrawer = ({ open, setOpen }: Props): JSX.Element => {
   const classes = useStyles();
+  const [boards, setBoards] = useState<Array<IBoard>>([]);
+  const { setActiveBoard } = useKanban();
+
+  const fetchBoard = async (id: string): Promise<IBoard> => {
+    const request = await getBoard(id);
+    setActiveBoard(request);
+    setOpen((prevOpen) => !prevOpen);
+
+    return request;
+  };
+
+  const getAllUserBoards = async () => {
+    const { boards } = await getUserBoards();
+    if (boards) setBoards(boards);
+
+    return undefined;
+  };
+
+  useEffect(() => {
+    getAllUserBoards();
+  }, []);
```

Comment:
Functionality for fetching the boards in the drawer rather than in the dashboard and passing it down; could consider having this live in context instead and could just grab the user's boards from it

Reply:
Agreed!